app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/actions/StoreLatestAction.scala

```diff
@@ -36,25 +39,38 @@ class StoreLatestActionImpl @Inject() (
     extends StoreLatestAction
     with Logging {
 
+  private def withTimeout[T](future: Future[T], timeout: FiniteDuration): Future[T] = {
+    implicit val system: ActorSystem = ActorSystem("TimeoutSystem")
```

Comment:
I would suggest injecting the `ActorSystem` instead of creating it. At the moment, this method creates an `ActorSystem` every time its called

Reply:
Yep, good point. I've pushed that up